great_expectations/checkpoint/checkpoint.py

```diff
@@ -520,6 +536,7 @@ def __init__(
         batches: Optional[List[dict]] = None,
         ge_cloud_id: Optional[UUID] = None,
         expectation_suite_ge_cloud_id: Optional[UUID] = None,
+        validation_id: Optional[str] = None,
```

Comment:
Don't love expand already verbose signatures but this is necessary due to the schema change - if we shift to a config-based API, we can minimize such issues.

Reply:
Not something that needs to be done right away but should we consider using a "parameter-object".
Like a `Options` type, or even more simply, an `Options` TypeDict`

Here are some relevant links, about how to refactor methods with lots of parameters. A `parameter-object` is one possible way to do it.

https://refactoring.guru/smells/long-parameter-list
https://refactoring.guru/introduce-parameter-object

Reply:
ah yeah the difficulty here is the fact that this is a very legacy bit of code that is at the forefront of the public API - I'm reluctant to make large scale changes here but perhaps we can implement something like you've noted in a non-breaking way.

Great website btw! I should read that more often.

Reply:
I want to understand this change better -- why do we have 3 different IDs for the same config?

Reply:
renamed `validation_id` to `default_validation_id` to better explain the purpose of the arg